server/metrics/metrics.go

```diff
@@ -202,6 +210,15 @@ func NewMetrics(info InstanceInfo) Metrics {
 	})
 	m.registry.MustRegister(m.linkedChannelsTotal)
 
+	m.upstreamUsersTotal = prometheus.NewGauge(prometheus.GaugeOpts{
+		Namespace:   MetricsNamespace,
+		Subsystem:   MetricsSubsystemApp,
+		Name:        "upstream_users_total",
```

Comment:
`_total` is a "special" suffix in Prometheus that's specific to a counter, but as this is a gauge, we should drop it in favour of just `upstream_users`.

Reply:
Proposing we rename the `Name` here as well:
```suggestion
		Name:        "upstream_users",
```